### ISSUE TITLE
Better tractor beam speed management

### ISSUE THREAD

**Issue by tibetiroka** [endless-sky/endless-sky#9759]
### Problem Description

Currently, a Tractor Beam's tractoring velocity is some base value divided by the tractored object's mass. While this is fine for asteroid mining, it does not scale well to heavy flotsam, such as engines.

Heavy flotsam is partially frozen in place (as seen from the ship) by the tractor beam. While this is not the intended use case, it still makes it a lot easier to retrieve the cargo, though only with sufficient maneuvering. Balance-wise this is an important distinction, because it helps more maneuverable ships more than less maneuverable ones that generally have a greater need for tractor beams.

### Related Issue Links

#9398

### Desired Solution

There could be a minimum velocity based on the tractor beam attribute that would be guaranteed even for the heaviest flotsam. Alternatively (or additionally), the scaling could be changed to be less aggressive; maybe similarly to how protection attributes work.

### Alternative Approaches

The tractor beam could be left as-is. This only affects factions that would want to scavenge flotsam from destroyed ships, and they are generally already capable of that without resorting to a tractor beam. Also, making the flotsam stationary is already helpful, even if it cannot be pulled in.

**Comment by Zitchas** [comment]
A combination of values cubes to mind: 
- Target mass (what size thing is this tractor beam most effective against)
- Target speed (how fast the ideal mass Target will be pulled in.)
- Below Mass Dropoff (how aggressive is the reduction in speed for things that weigh less)
- Over Mass Dropoff (how aggressive is the reduction in speed for things that are heavier)

This would allow a full range of variety, including being able to have it repel things.

**Comment by Amazinite** [comment]
Yeah. I'm thinking of changing it so that it takes the square or cubed root of the mass of the object being pulled, then reducing the strength of the current tractor beam by the same factor.

**Comment by Quotatron3000** [comment]
Just wanted to sound off on the tractor beam.

Currently, if an object is stationary, and your ship is making pace towards it, and the flotsam happens to be heavy, the tractor beam mechanics actually 'pushes it away' from you, making it take longer to retrieve or sometimes even malfunctioning and giving the flotsam additional unwanted acceleration.

The tractor beam needs to factor in the acceleration and angle of the ship itself also. So if I'm speeding towards the flotsam, it should include ship's acceleration + tractor pull rate to determine speed, not just tractor pull rate.
